### PR TITLE
go.mod: update pd client in 6.6

### PR DESCRIPTION
**Pull request by CabinfeverB**
Signed-off-by: Cabinfever_B <cabinfeveroier@gmail.com>

### What problem does this PR solve?

Issue Number: ref #38825 
should be merged after https://github.com/tikv/pd/pull/5981

Problem Summary:

### What is changed and how it works?
update pd client 
+ https://github.com/tikv/pd/pull/5981

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by glorv — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -274,5 +274,6 @@ replace (
 	// fix potential security issue(CVE-2020-26160) introduced by indirect dependency.
 	github.com/dgrijalva/jwt-go => github.com/form3tech-oss/jwt-go v3.2.6-0.20210809144907-32ab6a8243d7+incompatible
 	github.com/pingcap/tidb/parser => ./parser
+	github.com/tikv/pd/client => github.com/ti-chi-bot/pd/client v0.0.0-20230209034200-6d23a31c24be
```
Comment (hawkingrei) [inline go.mod:277]: Why does it change to use ```ti-chi-bot/pd```?

@nolouch

Comment (CabinfeverB) [reply go.mod:277]: I will change it after https://github.com/tikv/pd/pull/5981 merged

Comment (hawkingrei) [reply go.mod:277]: I hope it can be merged after https://github.com/tikv/pd/pull/5981 merged

**Comment by hawkingrei** [comment]
@CabinfeverB Please remember to run ```make bazel_prepare``` after updating gomod.

**Pull request merged**